### PR TITLE
Fix bare excepts

### PR DESCRIPTION
**Pull request by misabelber**
Reopen of this pull request (sorry!)

I tried to solve issue #93 and set a certain error in the bare excepts. I don't have a lot of experience in error handling in Python so I'm not completely sure if I used the proper error type. If anyone can take a look and see if seems correct, I will be thankful.

### CODE REVIEW

**Comment by rlopezcoto** [comment]
Thanks @misabelber! @dneise could you please have a look to this fix?

File: lstchain/io/lstcontainers.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def fill_hillas(self, hillas):
         for k in hillas.keys():
             try:
                 self[k] = hillas[k]
-            except:
+            except StopIteration:
```
Comment (dneise) [inline lstchain/io/lstcontainers.py]: I think StopItereation is wrong here.

File: lstchain/reco/dl0_to_dl1.py [hunk 2]
```diff
@@ -93,7 +93,7 @@ def get_dl1(calibrated_event, telescope_id):
             dl1_container.set_disp([dl1_container.src_x, dl1_container.src_y], hillas)
             return dl1_container
 
-        except:
+        except RuntimeError:
```
Comment (dneise) [inline lstchain/reco/dl0_to_dl1.py]: Which of the functions in this try...except block raises a `RuntimeError` ?

File: lstchain/reco/dl1_to_dl2.py [hunk 3]
```diff
@@ -106,7 +106,7 @@ def train_energy(train,
             config = read_configuration_file(config_file)
             regression_args = config['random_forest_regressor_args']
             print("Configuration loaded from {}".format(config_file))
-        except:
+        except OSError:
```
Comment (dneise) [inline lstchain/reco/dl1_to_dl2.py]: Why do you expect an OSError here?

**Comment by misabelber** [comment]
Haha, I tried and I failed! I guess I'm not understanding the error types...
I think maybe @vuillaut can help here. Thomas, I think you where who introduced most of these bare excepts during one of the refactorings of the code, do you know what kind of exceptions are going to be raised in each case? Thank you very much!

**Comment by rlopezcoto** [comment]
Solved in #104

**Pull request closed without merging**